src/firmware/main/world/firmware_ball.c

```diff
@@ -0,0 +1,51 @@
+#include "world/firmware_ball.h"
+
+#include <stdlib.h>
+
+struct FirmwareBall
+{
+    float (*get_ball_position_x)();
+    float (*get_ball_position_y)();
+    float (*get_ball_velocity_x)();
+    float (*get_ball_velocity_y)();
+};
+
+FirmwareBall* FirmwareBall_create(float (*get_ball_position_x)(),
+                                  float (*get_ball_position_y)(),
+                                  float (*get_ball_velocity_x)(),
+                                  float (*get_ball_velocity_y)())
+{
+    FirmwareBall* new_ball = malloc(sizeof(FirmwareBall));
+
+    new_ball->get_ball_position_x = get_ball_position_x;
```

Comment:
Nit:
I believe you were using 'new_ball->get_ball_position_x(get_ball_position_x)' in previous files. Change for consistancy?